### ISSUE TITLE
Burger menu should use <button> element with JS for accessibility

### ISSUE THREAD

**Issue by lexoyo** [silexlabs/Silex#1663]
The burger menu in all of our templates uses a CSS-only approach. For accessibility and also to make it easier to understand, it should be a `<button>` with proper JS toggle, and ARIA attributes

This is a work to do in a template, you'll need to [open a template (edit in silex button)](https://www.silex.me/templates/) then modify it and then click "Contribute changes" in the site settings:

<img width="1307" height="1117" alt="Silex site settings, with a Contribute changes link" src="https://github.com/user-attachments/assets/b66782ea-c1a9-4cb6-814c-5185ad1f4fa7" />
